SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -233,13 +223,28 @@ void CDiscAdjMultizoneDriver::Run() {
 
       Add_Solution_To_ExternalOld(iZone);
 
+      /*--- If the contents of BGSSolution are valid we initialize the inner iterations from them by moving
+       *    them to Solution, otherwise Solution contains only the contributions from the OF.
+       *    Note: This is not 100% correct as on the first inner iteration the updated cross terms
+       *          (from the previous zone) are missed. ---*/
+
+      if (iOuterIter > 0 || driver_config->GetRestart())
+        Set_Solution_To_BGSSolution(iZone);
+
```

Comment:
This is a semi-important change to have continuity from the last inner iteration to the first inner iteration of the next outer iteration. The solution in iZone gets overwritten when cross terms are extracted for another zone, so we need to restore from the BGS solution.